HLTrigger/HLTcore/interface/HLTPrescaleProvider.h

```diff
@@ -58,25 +59,53 @@ class HLTPrescaleProvider {
   // negative == error
 
   /// combining the two methods above
-  unsigned int prescaleValue(const edm::Event& iEvent, const edm::EventSetup& iSetup, const std::string& trigger);
-
+  template<typename T = unsigned int>
+  T prescaleValue(const edm::Event& iEvent, const edm::EventSetup& iSetup, const std::string& trigger){
+    const int set(prescaleSet(iEvent, iSetup));
+    return set < 0 ? -1 : 
+      hltConfigProvider_.prescaleValue<T>(static_cast<unsigned int>(set), trigger);
+  }
+  
   /// Combined L1T (pair.first) and HLT (pair.second) prescales per HLT path
-  std::pair<int, int> prescaleValues(const edm::Event& iEvent,
-                                     const edm::EventSetup& iSetup,
-                                     const std::string& trigger);
+  template<typename T = int>
+  std::pair<T, T> prescaleValues(const edm::Event& iEvent,
+				 const edm::EventSetup& iSetup,
+				 const std::string& trigger){
+    return {convertL1PS<T>(getL1PrescaleValue(iEvent,iSetup,trigger)),
+	prescaleValue<T>(iEvent,iSetup,trigger)};
```

Comment:
How does that create a std::pair to match the return type?

Reply:
I think that `return {x, y}` calls the constructor for `std::pair(x, y)`

Reply:
yes, that is my understanding.